apps/mobile/src/store/key-store.ts

```diff
@@ -109,6 +127,60 @@ export function useKeyStore() {
       const nextAccountIndex =
         fingerprintAccounts.length === 0 ? 0 : highestKeychainAccountIndex + 1;
 
+      // in extension secretKey is the mnemonic
+      const secretKey = mnemonic;
+
+      const legacyAccountActivityLookupGaia =
+        await checkForLegacyGaiaConfigWithKnownGeneratedAccountIndex(secretKey);
+
+      // hardcoding this to match extension count for dev wallet
+      // checkForLegacyGaiaConfigWithKnownGeneratedAccountIndex fails with
+      // [TypeError: _this.nodePbkdf2 is not a function (it is undefined)]
+      const legacyAccountActivityLookup =
+        legacyAccountActivityLookupGaia === 0 ? 123 : legacyAccountActivityLookupGaia;
+
+      async function doesStacksAddressHaveBalance(address: string) {
+        const controller = new AbortController();
+        const resp = await stxClient.getAccountBalance(address, controller.signal);
+        return Number(resp.stx.balance) > 0;
+      }
+
+      try {
+        // I think index here is not incrementing properly and it's needed by getStacksAddressByIndex
+        void recurseAccountsForActivity({
+          async doesAddressHaveActivityFn(index: number) {
+            const stxAddress = getStacksAddressByIndex(
+              secretKey,
+              AddressVersion.MainnetSingleSig
+            )(index);
+            // here we call doesStacksAddressHaveBalance which calls stacks client directly not using react query
+            const hasStxBalance = await doesStacksAddressHaveBalance(stxAddress);
+            // const hasNames = await doesStacksAddressHaveBnsName(stxAddress);
+            // const btcAddress = getNativeSegwitMainnetAddressFromMnemonic(secretKey)(index);
+            // const hasBtcBalance = await doesBitcoinAddressHaveBalance(btcAddress.address!);
+            // TODO: add inscription check here also?
+            console.log('doesAddressHaveActivityFn', index, stxAddress);
+            return hasStxBalance; // || hasNames || hasBtcBalance;
+          },
+        }).then(recursiveActivityIndex => {
+          if (recursiveActivityIndex <= legacyAccountActivityLookup) return;
+          console.info('Found account activity at higher index', { recursiveActivityIndex });
+          // dispatch(stxChainSlice.actions.restoreAccountIndex(recursiveActivityIndex));
+          console.log('recursiveActivityIndex', recursiveActivityIndex);
+          this.createNewAccountOfWallet(fingerprint, recursiveActivityIndex);
+        });
+      } catch {
+        // Errors during account restore are non-critical and can fail silently
+      }
+
+      // on the extension dev wallet this is where most accounts are created
+      if (legacyAccountActivityLookup !== 0)
```

Comment:
If I try restore one of my personal wallets on the `extenstion` the accounts also get generated here. 

That account is from 2023 at the earliest so I don't think it should be considered legacy?